code/modules/modular_armor/modular.dm

```diff
@@ -172,20 +173,32 @@
 	if(!istype(module))
 		return FALSE
 
-	if(ismob(loc) && (user.r_hand != src && user.l_hand != src))
-		if(!silent)
-			to_chat(user, span_warning("You need to remove the armor first."))
-		return FALSE
-
 	if(!do_after(user, equip_delay, TRUE, user, BUSY_ICON_GENERIC))
 		return FALSE
 
+///Temporarily unequips and then requips the armor during the attach process to keep armor datums/actions up to date if needed
+/obj/item/clothing/suit/modular/proc/do_attach(mob/living/user, obj/item/armor_module/module)
+	var/needs_reequip = user.get_item_by_slot(SLOT_WEAR_SUIT) == src
+	if(needs_reequip)
+		unequipped(user, SLOT_WEAR_SUIT)
+	module.do_attach(user, src)
+	if(needs_reequip)
+		equipped(user, SLOT_WEAR_SUIT)
+
```

Comment:
this does not actually use any extra utility of the suit this should be on the actual module
Additionally you named two procs the same leading to confusion but see above